modelcluster/fields.py

```diff
@@ -270,3 +271,278 @@ def check(self, **kwargs):
+
+            for item in final_items:
+                item.save()
+                original_manager.add(item)
+
+            # purge the _cluster_related_objects entry, so we switch back to live SQL
+            del self.instance._cluster_related_objects[relation_name]
+
+    return DeferringManyRelatedManager
+
+
+class ReverseM2MRelatedObjectsDescriptor(ReverseManyRelatedObjectsDescriptor):
+    @cached_property
+    def related_m2m_manager_cls(self):
+        # Dynamically create a class that subclasses the related model's
+        # default manager.
+        return create_deferring_many_related_manager(
+            self.field.rel,
+            self.related_manager_cls
+        )
+
+    def __get__(self, instance, instance_type=None):
+        if instance is None:
+            return self
+
+        return self.related_m2m_manager_cls(
+            model = self.field.rel.to,
+            query_field_name=self.field.related_query_name(),
+            prefetch_cache_name=self.field.name,
+            instance=instance,
+            symmetrical=self.field.rel.symmetrical,
+            source_field_name=self.field.m2m_field_name(),
+            target_field_name=self.field.m2m_reverse_field_name(),
+            reverse=False,
+            through=self.field.rel.through,
+        )
+
+    def __set__(self, instance, value):
+        manager = self.__get__(instance)
+        manager.clear()
+        manager.add(*value)
+
+
+class M2MRelatedObjectsDescriptor(ManyRelatedObjectsDescriptor):
+    @cached_property
+    def related_manager_cls(self):
+        # Dynamically create a class that subclasses the related
+        # model's default manager.
+        return create_deferring_many_related_manager(
+            self.related,
+            self.related_manager_cls
+        )
+
+    def __set__(self, instance, value):
+        manager = self.__get__(instance)
+        manager.clear()
+        manager.add(*value)
+
+
+class M2MField(ManyToManyField):
+    def contribute_to_class(self, cls, name, **kwargs):
+        super(M2MField, self).contribute_to_class(cls, name, **kwargs)
+        setattr(cls, self.name, ReverseM2MRelatedObjectsDescriptor(self))
+        self.field = getattr(cls, self.name)
+
+    def contribute_to_related_class(self, cls, related):
+        if not self.rel.is_hidden() and not get_related_model(related)._meta.swapped:
+            setattr(cls, related.get_accessor_name(), M2MRelatedObjectsDescriptor(related))
+        try:
+            related.model._meta.child_relations.append(related)
+        except AttributeError:
+            related.model._meta.child_relations = [related]
+        super(M2MField, self).contribute_to_related_class(cls, related)
+
+    def get_accessor_name(self):
+        return self.name
+
+    def _get_val_from_obj(self, obj):
+        mgr = super(M2MField, self)._get_val_from_obj(obj)
+        return mgr.get_object_list()
+
+    def get_searchable_content(self, value):
```

Comment:
I think this should be left out. `get_searchable_content` is only used by Wagtail's search mechanism, and we should keep Wagtail-specific functionality out of modelcluster.

In any case, indexing a list of PKs probably isn't going to be very useful for searching (and it's hard to know whether indexing the actual content of the related model would be a good thing or not... see the discussion at https://github.com/torchbox/wagtail/issues/2500#issuecomment-212158853). Hopefully [`index.RelatedFields`](http://docs.wagtail.io/en/v1.4.3/topics/search/indexing.html#index-relatedfields) already handles M2M relations - or can be updated to do so - which would mean we don't need any special treatment of search indexing here.